flake8_builtins.py

```diff
@@ -140,7 +140,9 @@ def run(self):
 
     def check_assignment(self, statement):
         msg = self.assign_msg
-        if type(statement.__flake8_builtins_parent) is ast.ClassDef:
+        if isinstance(statement.__flake8_builtins_parent, ast.ClassDef):
+            if "TypedDict" in {a.id for a in statement.__flake8_builtins_parent.bases if isinstance(a, ast.Name)}:
```

Comment:
@danjones1618 do we expect other typing classes to be used eventually here? 🤔 

Given how long this conditional is, it might be worth refactoring into a function?

```python
def base_classes_ids(self, statement):
    return {a.id for a in statement.__flake8_builtins_parent.bases if isinstance(a, ast.Name)}
```